code/datums/action.dm

```diff
@@ -107,8 +107,10 @@
 
 		ApplyIcon(button)
 
+		var/obj/effect/proc_holder/spell/S = target
+
 		// If the action isn't available, darken the button
-		if(!IsAvailable())
+		if(!IsAvailable() || istype(S) && S.cooldown_handler.get_availability_percentage() != 1)
 			apply_unavailable_effect()
```

Comment:
dont do this, 
add this to the override of `/datum/action/spell_action/IsAvailable()`

Reply:
if i do what you said, clicking spell will be impossible until IsAvailable return TRUE
if i return TRUE when spell is avaiable, there will be no progress and 1/2 on spell icon
here i mean that spell is avaiable, but also we want darker icon to show recharging process and its status

this fella would block using it
![image](https://github.com/ParadiseSS13/Paradise/assets/61080616/01c7aedd-8393-46ee-819a-8a8b84032d84)

Reply:
Placed other proc here, new one ShouldShowCooldown
Should be fine now
https://github.com/ParadiseSS13/Paradise/pull/21065/commits/465eb21a3835d5cecd2295822383c7eca4b32c17